src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage.cs

```diff
@@ -243,6 +243,9 @@ public override void Initialize(Solution solution)
             {
                 var connection = pooledConnection.Connection;
 
+                // Enable WAL
+                connection.ExecuteCommand("pragma journal_mode=wal", throwOnError: false);
```

Comment:
❓ What does the performance look like if you use `journal_mode=memory` instead?

Reply:
Didn’t even consider it since devenv/servicehub crash might corrupt DB and make it unusable. Not sure this is an option worth exploring given the possibility of corrupting data. I understand that stored data is transient, but I’m afraid that subsequent operations might crash devenv/servicehub permanently, requiring users to manually delete the DBs as a fix.

Reply:
We could create a file in the same directory, and delete it if/when the service shuts down cleanly. The state of that file would tell us if the database can be trusted:

* File does not exist: database is fine
* File exists and is not locked: application terminated unexpectedly; delete the file and the database to reset
* File exists and is locked: application is still running; database cannot be used

Reply:
:memo: `journal_mode=memory` should provide at least twice the benefit we would get by moving to `journal_mode=wal`. Given the data is transient, I believe it's well worth the effort to see if we can leverage the in-memory setting by finding a way to detect possible corruption.

Reply:
@sharwell Sure, I'll take a look both on the performance and on the failure modes.

P.S. sorry for the delay, was travelling, 15-hour flights are no fun.

Reply:
I feel like trying to detect corruption could be very error prone.  We could easily have bugs there that go undedected and which lead to all sorts of bad transient experiences in the IDE. 

WAL seems especially reasonable.  It's a good default for the types of read/write patterns we have, and it doesn't relax the strong db integrity and consistency guarantees that we have designed for.

Reply:
Note: i would not describe the data as being "transient". Indeed, it's the opposite.  The intent of the db is (nearly all of the time) to allow us to compute data once and then store it for exceptionally long periods of time (days, weeks, months, etc.).   Until the data actually needs to be updated, we want to store what we have across sessions to prevent unnecessary recomputation of it.  As most data is not actually changed, this is an appropriate design that balances the goals of fast index updating, with low overall CPU utilization.  

As mentioned above, I would stick with the DB approach that still provides strong consistency and integrity, and I would see what could be tweaked within those boundaries to make it not a bottleneck.

Reply:
I would also note, that the original cause of "slowness" described was precisely because SQLite tested a file to exist on each operation. While I understand that this can be done in Initialize/Dispose phase instead of per-transaction, but exclusive locking will block multiple instances of servicehub/devenv of using the DB simultaneously.

Reply:
@CyrusNajmabadi By "transient" I meant that it's not a primary data source, merely a cache. From a performance standpoint it's of course very important to preserve it if possible.

Reply:
> exclusive locking will block multiple instances of servicehub/devenv of using the DB simultaneously

It's my understanding that we already placed this limitation in #21532.

Reply:
@sharwell my bad, thanks for pointing this out.